src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php

```diff
@@ -25,8 +25,12 @@ public function __construct(
     ) {
     }
 
-    public function vote(TokenInterface $token, mixed $subject, array $attributes): int
+    /**
+     * @param Vote $vote Should be used to explain the vote
+     */
+    public function vote(TokenInterface $token, mixed $subject, array $attributes/* , Vote $vote = new Vote() */): int
     {
+        $vote = 3 < \func_num_args() ? func_get_arg(3) : new Vote();
```

Comment:
I like this :)

Reply:
Why is func_num_args always globally namespaced here, but func_get_arg is not?

Reply:
We are namespacing only the functions that have a compiler optimization when they are fully-qualified. `func_num_args` does while `func_get_args` does not.